provision/acc_provision/acc_provision.py

```diff
@@ -950,11 +931,88 @@ def provision(args, apic_file, no_random):
     return ret
 
 
+DEFAULT_FLAVORS_PATH = os.path.dirname(os.path.realpath(__file__)) + "/flavors.yaml"
+VERSION_FIELDS = [
+    "cnideploy_version",
+    "aci_containers_host_version",
+    "opflex_agent_version",
+    "aci_containers_controller_version",
+    "openvswitch_version",
+]
+
+
+def check_yaml(text_to_verify):
+    try:
+        flavors_yaml = yaml.load(text_to_verify)
+        if 'default_flavor' in flavors_yaml:
+            return flavors_yaml
+    except yaml.YAMLError as exc:
+        print(exc)
+        raise Exception
+
+
+def default_flavors_path():
+    with open(DEFAULT_FLAVORS_PATH, 'r') as res:
+        info("Loading flavors from default flavors.yaml: " + DEFAULT_FLAVORS_PATH)
+        valid_yaml = check_yaml(res)
+        if valid_yaml:
+            set_flavors(valid_yaml)
+    return
+
+
+def get_flavors(flavors_url):
+    if flavors_url == 'blah':
+        return default_flavors_path()
+    flag = False
+    try:
+        # try as a URL
+        res = requests.get(flavors_url)
+        if res and res.status_code == 200:
+            valid_yaml = check_yaml(res.text)
+            if valid_yaml:
+                info("Loading flavors from URL: " + flavors_url)
+                set_flavors(valid_yaml)
+                flag = True
+        else:
+            raise Exception
```

Comment:
can we raise specific Exception ? Catching generic exception is not good.

Reply:
Should I raise one of the built-in exceptions? Or would defining custom be better because I need to catch both bad URL and bad YAML errors

Reply:
Included in the latest commit.